### PR TITLE
Update build OCK action

### PR DESCRIPTION
**Pull request by alan-forbes-cp**
# Overview

Update build OCK action for new MR jobs. (A corresponding PR for the first such job will follow once this is merged)

# Reason for change

Running a range of selected MR jobs in my Github fork has shown the need to update and expand the build action. The MR jobs are now also running without the build.py script previously used in Gitlab.

# Description of change

General approach has been to review existing settings and add new ones as required with a reasonable default.

The action has been tested against a range of MR jobs in my fork.

# Anything else we should know?

Changes and further additions are highly likely as more jobs are run.

### CODE REVIEW

File: .github/actions/do_build_ock/action.yml [hunk 1]
```diff
@@ -73,44 +88,80 @@ inputs:
...
+              -DCMAKE_C_COMPILER_LAUNCHER=sccache \
+              -DCMAKE_CXX_COMPILER_LAUNCHER=sccache                 \
+              -DCA_EXTERNAL_ONEAPI_CON_KIT_DIR=${{ inputs.oneapi_con_kit_dir }} \
+              -DCA_MUX_TARGETS_TO_ENABLE=${{ inputs.mux_targets_enable }} \
+              -DCA_ENABLE_API=${{ inputs.enable_api }} \
+              -DCA_LLVM_INSTALL_DIR=${{ inputs.llvm_install_dir }} \
+              -DCA_ENABLE_DEBUG_SUPPORT=${{ inputs.debug_support }} \
+              -DCMAKE_BUILD_TYPE=${{ inputs.build_type }} \
+              -DCA_ENABLE_HOST_IMAGE_SUPPORT=${{ inputs.host_image }} \
+              -DCA_HOST_ENABLE_BUILTIN_KERNEL=${{ inputs.debug_support }} \
```
Comment (alan-forbes-cp) [inline .github/actions/do_build_ock/action.yml:135]: This is a legacy setting - but it looks wrong to me. Any comments?

File: .github/actions/do_build_ock/action.yml [hunk 2]
```diff
@@ -4,15 +4,18 @@ description: Action to build the oneapi-construction-kit
...
     description: 'build type e.g Release, ReleaseAssert (default ReleaseAssert)'
     default: ReleaseAssert
   mux_targets_enable:
     description: 'mux targets to enable e.g. riscv, host (default host)'
     default: "host"
   mux_compilers_enable:
-    description: 'mux compiler target to enable e.g. riscv, host (default host)'
-    default: "host"
+    description: 'mux compiler target to enable e.g. riscv, host (no default)'
+    default: ""
```
Comment (alan-forbes-cp) [inline .github/actions/do_build_ock/action.yml:18]: Harald raised the issue of the default setting for this one so comments welcome on any default value needed.

File: .github/actions/do_build_ock/action.yml (hunk 1 above)
Comment (coldav) [reply .github/actions/do_build_ock/action.yml:135]: does look wrong yes

File: .github/actions/do_build_ock/action.yml (hunk 2 above)
Comment (coldav) [reply .github/actions/do_build_ock/action.yml:18]: I'm okay with the change to ""

File: .github/actions/do_build_ock/action.yml (hunk 1 above)
Comment (alan-forbes-cp) [reply .github/actions/do_build_ock/action.yml]: Done

File: .github/actions/do_build_ock/action.yml [hunk 3]
```diff
@@ -44,11 +50,23 @@ inputs:
     description: 'External compiler directories, default is to empty string, which means no external directories'
     default: ""
   hal_description:
-    description: 'Description to be used for HAL, typically risc-v ISA (default RV64GCV)'
-    default: "RV64GCV"
+    description: 'Description to be used for HAL, typically risc-v ISA (default "")'
+    default: ""
+  refsi_tutorial_hal_dir:
```
Comment (coldav) [inline .github/actions/do_build_ock/action.yml:55]: I feel refsi_tutorial are too bespoke to a job and should be removed (and done via the extra args). Same goes for oneapi_con_kit_dir.
Note sure about hal_refsi_thread_mode, maybe that is okay.

Comment (alan-forbes-cp) [reply .github/actions/do_build_ock/action.yml:55]: I'd say keep `hal_refsi_thread_mode` as its used in multiple jobs with different non-default values.
I'll update for the other two you mention.
Ta.

Comment (coldav) [reply .github/actions/do_build_ock/action.yml:55]: to be clear there are two refsi_tutorial ones and one con_kit one (all related to refsi tutorial).

**Review by coldav — APPROVED** [review]

File: .github/actions/do_build_ock/action.yml (hunk 3 above)
Comment (alan-forbes-cp) [reply .github/actions/do_build_ock/action.yml]: These 3 inputs are now deleted.

**Review by coldav — APPROVED** [review]

**Pull request merged**